src/main/java/org/alliancegenome/curation_api/dao/AlleleDAO.java

```diff
@@ -68,4 +80,49 @@ public Boolean hasReferencingPhenotypeAnnotations(Long alleleId) {
 		results = agmPhenotypeAnnotationDAO.findIdsByParams(agmPaParams);
 		return CollectionUtils.isNotEmpty(results);
 	}
+
+	public SearchResponse<Allele> findAllAllelesWithConstructs(Pagination pagination, HashMap<String, Object> params) {
```

Comment:
Is the reason for this method because the generic method doesn't have a "is not empty" way of querying?

Reply:
@cmpich can we add that in a generic way as in

params.put("field", ""): or params.put("field", null):

Then check on the other size and if its empty string or null added the correct condition?